### ISSUE TITLE
Can’t load `global` in workers with scriptLoad=true

### ISSUE THREAD

**Issue by surma** [systemjs/systemjs#1322]
I am using [idb.js](https://www.npmjs.com/package/idb) inside my ServiceWorker but can’t load it using SystemJS because [`webWorkerImport()`](https://github.com/systemjs/systemjs/blob/f94507b22bc5aff669e9d404d43a6bd6e60fcf42/lib/scriptLoader.js#L77) expects an AMD or SystemJS module, regardless of the `format` config field.

As a workaround, I am currently doing this inside my ServiceWorker (not very generic code, tho):

```JS
SystemJS.config({
    "meta": {
      "*": {
        "scriptLoad": true
      },
      "/idb.js": {
        "scriptLoad": true,
        "format": "global",
        "exports": "idb"
      }
    }
  })

const oldFetch = System.fetch;
System.fetch = function(load) {
  if(load.metadata.scriptLoad && load.metadata.format === 'global') {
    importScripts(load.address);
    load.metadata.entry = self[load.metadata.exports];
    return Promise.resolve('');
  }
  return oldFetch.call(this, load);
};
```

Is there anything I am missing? I haven’t quite wrapped my hand about SystemJS’ architecture, otherwise I’d have whipped up a PR right away, provided it’s appropriate.

**Comment by surma** [comment]
I forgot to restrict this to `scriptLoad: true` only, which is important for ServiceWorker to make sure the dependencies get cached.

**Comment by guybedford** [comment]
I've added a possible fix in https://github.com/systemjs/systemjs/commit/788c3fb72247b4b97bb2300e94eebc850f43ced8. Let me know if that works out for you here?

**Comment by surma** [comment]
Haven’t actually tested the code, but from the looks of it that would be good enough for my use-case! Great work, thanks a lot :)

**Comment by guybedford** [comment]
Sure no problem, will keep as fixed pending release for now.

**Comment by guybedford** [comment]
Released in 0.19.34.
